tests/integration/targets/aci_igmp_interface_policy/tasks/main.yml

```diff
@@ -0,0 +1,239 @@
+# Test code for the ACI modules
+# Copyright: (c) 2023, Tim Cragg (@timcragg)
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact:
+   aci_info: &aci_info
+    host: "{{ aci_hostname }}"
+    port: "{{ aci_port | default(omit) }}"
+    username: "{{ aci_username }}"
+    password: "{{ aci_password }}"
+    validate_certs: '{{ aci_validate_certs | default(false) }}'
+    use_ssl: '{{ aci_use_ssl | default(true) }}'
+    use_proxy: '{{ aci_use_proxy | default(true) }}'
+    output_level: '{{ aci_output_level | default("info") }}'
+
+# CLEAN ENVIRONMENT
+- name: Remove the ansible_tenant
+  cisco.aci.aci_tenant:
+    <<: *aci_info
+    name: ansible_tenant
+    state: absent
+
+# CREATE IGMP INTERFACE POLICY
+- name: Create the ansible_tenant
+  cisco.aci.aci_tenant:
+    <<: *aci_info
+    name: ansible_tenant
+    state: present
+
+- name: Create an IGMP interface policy (check mode)
+  cisco.aci.aci_igmp_interface_policy: &aci_igmp
+    <<: *aci_info
+    name: ansible_igmp_intf_policy
+    tenant: ansible_tenant
+    description: Test IGMP Interface Policy
+    group_timeout: 300
+    query_interval: 100
+    query_response_interval: 8
+    last_member_count: 3
+    last_member_response: 5
+    startup_query_count: 3
+    startup_query_interval: 5
+    querier_timeout: 200
+    robustness_variable: 3
+    igmp_version: v3
+    allow_v3_asm: true
```

Comment:
from the coverage report noticed that `allow_v3_asm: false` is never tested could you add this scenario in test case.

Reply:
Updated tests so allow_v3_asm state is changed during "Update IGMP Interface Policy" task